File: main/symmetric-tree/symmetric-tree-iterative.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public bool IsSymmetric(TreeNode root) {
        if (root == null) return true;
        var ls = new Stack<TreeNode>();
        var rs = new Stack<TreeNode>();
        ls.Push(root.left);
        rs.Push(root.right);

        while (ls.Count != 0) {
            var left = ls.Pop();
            var right = rs.Pop();

            if (left == null) {
                if (right != null) return false;
                continue;
            }

            if (right == null || left.val != right.val) return false;

            ls.Push(left.left);
            ls.Push(left.right);
            rs.Push(right.right);
            rs.Push(right.left);
        }

        return true;
    }
}

File: main/symmetric-tree/symmetric-tree-iterative.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    @staticmethod
    def isSymmetric(root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        if root is None:
            return True

        stack = [(root.left, root.right)]
        while stack:
            left, right = stack.pop()
            if left is None or right is None:
                if left is not None or right is not None:
                    return False
            else:
                if left.val != right.val:
                    return False
                stack.append((left.right, right.left))
                stack.append((left.left, right.right))

        return True
